BDArmory/Competition/VesselSpawning/WaypointCourses.cs

```diff
@@ -0,0 +1,263 @@
+            {
+                foreach (var location in defaultLocations.ToList())
+                    if (!WaypointCourses.CourseLocations.Select(l => l.name).ToList().Contains(location.name))
+                        WaypointCourses.CourseLocations.Add(location);
+            }
+        }
+
+        public static object ParseValue(Type type, string value)
+        {
+            try
+            {
+                if (type == typeof(string))
+                {
+                    return value;
+                }
+                else if (type == typeof(bool))
+                {
+                    return Boolean.Parse(value);
+                }
+                else if (type == typeof(int))
+                {
+                    return int.Parse(value);
+                }
+                else if (type == typeof(float))
+                {
+                    return float.Parse(value);
+                }
+                else if (type == typeof(Vector2))
+                {
+                    char[] charsToTrim = { '(', ')', ' ' };
+                    string[] strings = value.Trim(charsToTrim).Split(',');
+                    if (strings.Length == 2)
+                    {
+                        float x = float.Parse(strings[0]);
+                        float y = float.Parse(strings[1]);
+                        return new Vector2(x, y);
+                    }
+                }
+                else if (type == typeof(Vector3))
+                {
+                    char[] charsToTrim = { '[', ']', '(', ')', ' ' };
+                    string[] strings = value.Trim(charsToTrim).Split(',');
+                    float x = float.Parse(strings[0]);
+                    float y = float.Parse(strings[1]);
+                    float z = float.Parse(strings[2]);
+                    return new Vector3(x, y, z);
+                }
+                else if (type == typeof(WaypointCourse))
+                {
+                    string[] parts;
+
+                    parts = value.Split(new char[] { ';' });
+                    if (parts.Length > 1)
+                    {
+                        var name = (string)ParseValue(typeof(string), parts[0]);
+                        var worldIndex = (int)ParseValue(typeof(int), parts[1]);
+                        var spawnPoint = (Vector2)ParseValue(typeof(Vector2), parts[2]);
+                        string[] waypoints = parts[3].Split(new char[] { ':' });
+                        List<Waypoint> waypointList = new List<Waypoint>();
+                        for (int i = 0; i < waypoints.Length-1; i++)
+                        {
+                            string[] datavars;
+                            datavars = waypoints[i].Split(new char[] { '|' });
+                            string WPname = (string)ParseValue(typeof(string), datavars[0]);
+                            WPname = WPname.Trim(' ');
+                            var location = (Vector3)ParseValue(typeof(Vector3), datavars[1]);
```

Comment:
Adding ``if (string.IsNullOrEmpty(WPname)) WPname = $"Waypoint ";``  on ln 244 would default blank names to 'Waypoint n';